pkg/reporter/gitlab.go

```diff
@@ -0,0 +1,358 @@
+// following GitLab's identifier guidelines
+// Docs: https://docs.gitlab.com/development/integrations/secure/#identifiers
+func gitlabAddVulnerabilityIdentifiers(vuln *gitLabVulnerability, vulnData *insightapi.PackageVulnerability) {
+	// Extract identifiers from the vulnerability data
+	identifiersFound := make(map[gitLabIdentifierType][]string)
+	aliases := utils.SafelyGetValue(vulnData.Aliases)
+
+	for _, alias := range aliases {
+		switch {
+		case strings.HasPrefix(alias, "CVE-"):
+			identifiersFound[gitLabIdentifierTypeCVE] = append(identifiersFound[gitLabIdentifierTypeCVE], alias)
+		case strings.HasPrefix(alias, "CWE-"):
+			identifiersFound[gitLabIdentifierTypeCWE] = append(identifiersFound[gitLabIdentifierTypeCWE], alias)
+		case strings.HasPrefix(alias, "GHSA-"):
+			identifiersFound[gitLabIdentifierTypeGHSA] = append(identifiersFound[gitLabIdentifierTypeGHSA], alias)
+		}
+	}
+
+	// Priority order of identifiers
+	// Since we can only who {gitlabMaxIdentifiers} in gitlab, we need to prioritize identifiers
+	identifiersPriority := []gitLabIdentifierType{
+		gitLabIdentifierTypeCVE,
+		gitLabIdentifierTypeCWE,
+		gitLabIdentifierTypeGHSA,
+	}
+
+	// Add identifiers in order of priority to report
+	reportIdentifiers := make([]gitLabIdentifier, 0)
+
+	for _, idfsType := range identifiersPriority {
+		for _, identifier := range identifiersFound[idfsType] {
+			reportIdentifiers = append(reportIdentifiers, gitLabIdentifier{
+				Type:  idfsType,
+				Name:  identifier,
+				Value: identifier,
+				URL:   common.GetIdentifierURL(string(idfsType), identifier),
+			})
+		}
+	}
+
+	// If identifiers are more than {gitlabMaxIdentifiers}, then system saves only {gitlabMaxIdentifiers}, so why increase the network cost
+	if len(reportIdentifiers) > gitlabMaxIdentifiers {
+		reportIdentifiers = reportIdentifiers[:gitlabMaxIdentifiers]
+	}
+
+	vuln.Identifiers = reportIdentifiers
+}
+
+func (r *gitLabReporter) AddManifest(manifest *models.PackageManifest) {
+	// Process each package in the manifest
+	for _, pkg := range manifest.Packages {
+		if pkg.Insights == nil {
+			continue
+		}
+
+		// Package location
+		location := gitLabLocation{
+			File: manifest.Path,
+			Dependency: gitLabDependency{
+				Package: gitLabPackage{
+					Name: pkg.GetName(),
+				},
+				Version: pkg.GetVersion(),
+				Direct:  pkg.IsDirect(),
+			},
+		}
+
+		// Add malware analysis result
+		malwareAnalysis := pkg.MalwareAnalysis
+
+		if malwareAnalysis != nil && (malwareAnalysis.IsMalware || malwareAnalysis.IsSuspicious) {
+			severity := SeverityCritical
+			if malwareAnalysis.IsSuspicious {
+				severity = SeverityHigh
+			}
+
+			description := ""
+			reportUrl := ""
+
+			if malwareAnalysis.Report != nil {
+				reportUrl = malysis.ReportURL(malwareAnalysis.Report.ReportId)
+				if malwareAnalysis.Report.Inference != nil {
+					description = fmt.Sprintf("%s\n\n%s", malwareAnalysis.Report.Inference.Summary, malwareAnalysis.Report.Inference.Details)
```

Comment:
Here we can use `malwareAnalysis.Report.GetInference().GetSummary()` to be nil safe. Same for `Inference.Details`